Content.Shared/Cargo/CargoBountyHistoryData.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Shared.Cargo.Prototypes;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Cargo;
+
+/// <summary>
+/// A data structure for storing historical information about bounties.
+/// </summary>
+[DataDefinition, NetSerializable, Serializable]
+public readonly partial record struct CargoBountyHistoryData
+{
+    /// <summary>
+    /// A unique id used to identify the bounty
+    /// </summary>
+    [DataField]
+    public string Id { get; init; } = string.Empty;
+
+    /// <summary>
+    /// Whether this bounty was completed or skipped.
+    /// </summary>
+    [DataField]
+    public BountyResult Result { get; init; } = BountyResult.Completed;
+
+    /// <summary>
+    /// Optional name of the actor that completed/skipped the bounty.
+    /// </summary>
+    [DataField]
+    public string? ActorName { get; init; } = default;
+
+    /// <summary>
+    /// Time when this bounty was completed or skipped
+    /// </summary>
+    [DataField]
+    public TimeSpan Timestamp { get; init; } = TimeSpan.MinValue;
+
+    /// <summary>
+    /// The prototype containing information about the bounty.
+    /// </summary>
+    [DataField(required: true)]
+    public ProtoId<CargoBountyPrototype> Bounty { get; init; } = string.Empty;
+
+    public CargoBountyHistoryData(CargoBountyData bounty, BountyResult result, TimeSpan timestamp, string? actorName)
+    {
+        Bounty = bounty.Bounty;
+        Result = result;
+        Id = bounty.Id;
+        ActorName = actorName;
+        Timestamp = timestamp;
+    }
+
+    /// <summary>
+    /// Covers how a bounty was actually finished.
+    ///     Completed - Bounty was actually fulfilled and the goods sold
+    ///     Skipped - Bounty was explicitly skipped by some actor
+    /// </summary>
+    public enum BountyResult
+    {
+        Completed = 0,
+        Skipped = 1,
+    }
```

Comment:
You can add documentation directly to the enum members for better ide support

```suggestion
    /// <summary>
    /// How a bounty was actually finished.
    /// </summary>
    public enum BountyResult
    {
        /// <summary>
        /// Bounty was completed and sold.
        /// </summary>
        Completed = 0,

        /// <summary>
        /// Bounty was skipped.
        /// </summary>
        Skipped = 1,
    }
```